module/VuFind/src/VuFind/Db/Service/ResourceTagsService.php

```diff
@@ -158,27 +158,64 @@ public function deleteLinksByResourceTagsIdArray(array $ids): int
     /**
      * Unlink tag rows for the specified resource and user.
      *
-     * @param int|int[]|null                               $resourceId ID (or array of IDs) of resource(s) to
+     * @param int|int[]|null                   $resourceId ID (or array of IDs) of resource(s) to
      * unlink (null for ALL matching resources)
-     * @param UserEntityInterface|int                      $userOrId   ID or entity representing user
-     * @param UserListEntityInterface|int|bool|string|null $listOrId   ID of list to unlink (null for ALL matching
-     * tags, 'none' for tags not in a list, true for tags only found in a list)
-     * @param int|int[]|null                               $tagId      ID or array of IDs of tag(s) to unlink (null
+     * @param UserEntityInterface|int          $userOrId   ID or entity representing user
+     * @param UserListEntityInterface|int|null $listOrId   ID of list to unlink (null for ALL matching tags)
+     * @param int|int[]|null                   $tagId      ID or array of IDs of tag(s) to unlink (null
      * for ALL matching tags)
      *
      * @return void
      */
     public function destroyResourceTagsLinksForUser(
         int|array|null $resourceId,
         UserEntityInterface|int $userOrId,
-        UserListEntityInterface|int|bool|string|null $listOrId = null,
+        UserListEntityInterface|int|null $listOrId = null,
         int|array|null $tagId = null
-    ) {
+    ): void {
         $userId = $userOrId instanceof UserEntityInterface ? $userOrId->getId() : $userOrId;
         $listId = $listOrId instanceof UserListEntityInterface ? $listOrId->getId() : $listOrId;
         $this->getDbTable('ResourceTags')->destroyResourceLinks($resourceId, $userId, $listId, $tagId);
     }
 
+    /**
+     * Unlink tag rows that are not associated with a favorite list for the specified resource and user.
+     *
+     * @param int|int[]|null          $resourceId ID (or array of IDs) of resource(s) to unlink (null for ALL matching
+     * resources)
+     * @param UserEntityInterface|int $userOrId   ID or entity representing user
+     * @param int|int[]|null          $tagId      ID or array of IDs of tag(s) to unlink (null for ALL matching tags)
+     *
+     * @return void
+     */
+    public function destroyNonListResourceTagsLinksForUser(
```

Comment:
As far as I can tell, this is never used, but since the functionality previously existed, I thought it was worth putting it in the interface.